### PR TITLE
Components: Fix highlighter scrolling bug

### PR DESCRIPTION
**Pull request by asanehisa**
When scrolling in either the Preview Panel or outside in the full page, clicking on components caused the Highlighter in the Preview Panel to not match up with the Component itself and instead be offset by the scroll. 

It should be offset by the preview panel scroll (after the preview panel iFrame issues are resolved in 2857). 

Must be merged after [SLAP-2857](https://github.com/yext/studio-prototype/pull/310)

J=SLAP-2858
TEST=auto, manual

Run local version + ensure highlighter issues are gone after changes. [Loom](https://www.loom.com/share/2acbd1466d2d4e10ac8570fbd138e797?sid=fc79433b-2871-4d27-a0ff-d1f4d0afbfa3)

### CODE REVIEW

File: packages/studio/src/components/Highlighter.tsx [hunk 1]
```diff
@@ -23,11 +23,16 @@ export default function Highlighter() {
     if (!rect) {
       return {};
     }
+    const IFrame = document.querySelector("iframe");
```
[the remark this answers is not in the document]
Comment (asanehisa) [reply packages/studio/src/components/Highlighter.tsx:26]: Hmm the highlighter is in the iframe I believe? Not 100% sure this is what you meant, but if I call window.scrollY inside the iframe I still get the full pages's scrollY not the IFrame's.

[the remark this answers is not in the document]
Comment (oshi97) [reply packages/studio/src/components/Highlighter.tsx:26]: Oh I thought this document would be the iframe's document but it's not

Comment (oshi97) [inline packages/studio/src/components/Highlighter.tsx:26]: nit lowercase i

Comment (oshi97) [inline packages/studio/src/components/Highlighter.tsx:26]: could we pass this iframe via react context (or something similar) instead? we already have a ref for it in IFramePortal

Comment (asanehisa) [reply packages/studio/src/components/Highlighter.tsx]: Oooh great idea

File: packages/studio/src/components/IFramePortal.tsx [hunk 2]
```diff
@@ -1,22 +1,23 @@
 import { createPortal } from "react-dom";
-import { CSSProperties, PropsWithChildren, useEffect, useState } from "react";
+import { CSSProperties, PropsWithChildren, useEffect } from "react";
 
 export default function IFramePortal(
   props: PropsWithChildren<{
     className?: string;
     title: string;
     inlineStyles: CSSProperties;
+    iframeEl;
```
Comment (oshi97) [inline packages/studio/src/components/IFramePortal.tsx:9]: add types

File: packages/studio/src/components/Highlighter.tsx [hunk 3]
```diff
@@ -6,7 +6,8 @@ import { ComponentStateKind } from "@yext/studio-plugin";
 /**
  * Highlights the current activeComponentRect on the page.
  */
-export default function Highlighter() {
+export default function Highlighter(props) {
+  const { iframeEl } = props;
```
Comment (oshi97) [inline packages/studio/src/components/Highlighter.tsx:10]: add interface for props

File: packages/studio/src/components/Highlighter.tsx [hunk 4]
```diff
@@ -26,15 +29,21 @@ export default function Highlighter() {
     return {
       position: "absolute",
       zIndex: "10",
-      left: `${window.scrollX + rect.left}px`,
-      top: `${window.scrollY + rect.top}px`,
+      left: `${
+        (iframeEl?.contentWindow ? iframeEl.contentWindow.scrollX : 0) +
```
Comment (oshi97) [inline packages/studio/src/components/Highlighter.tsx:33]: nit but this (and below) can be `iframe?.contentWindow?.scrollX ?? 0`

Comment (asanehisa) [reply packages/studio/src/components/Highlighter.tsx:33]: Oooh ur right! Thanks!

**Review by oshi97 — APPROVED** [review]
lgtm!

**Pull request merged**